src/ui/page/share/dataset/steps/serviceStorageSelection/serviceStorageSelection.scss

```diff
@@ -28,31 +33,45 @@
         letter-spacing: 0.4px;
         color: themed('text');
         margin-bottom: 16px;
+        text-transform: uppercase;
       }
     }
 
+    .okp4-dataverse-portal-share-dataset-page-service-selection-search-container {
+      @include grid-container($colGutter: 18px, $rowGutter: $search-container-gap);
+    }
+    
     .okp4-dataverse-portal-searchbar-main {
-      @include grid-item($colStart: 1, $colEnd: 7, $rowStart: auto, $rowEnd: auto);
+      @include grid-item($colStart: 1, $colEnd: 7, $rowStart: 1, $rowEnd: 1);
     }
 
     .okp4-dataverse-portal-share-dataset-page-services-container {
-      @include grid-item($colStart: 1, $colEnd: -1, $rowStart: auto, $rowEnd: auto);
-      max-height: 420px;
+      box-sizing: border-box;
+      position: relative; // necessary to compute the height of the scroll container
+      padding-bottom: $search-container-gap;
+      overflow-y: auto;
+      @include grid-item($colStart: 1, $colEnd: -1, $rowStart: 2, $rowEnd: 16);
```

Comment:
It is a simple way to manage spacing between row elements, but probably not the cleanest one. Do you have any recommendation on managing spacing in this context?

Reply:
Yes I think you could do something like this (I tried it and seems ok):
- you keep the grid on the search container
- you create one container with searchBar and services display, which will be positioned in the grid mentioned above, and add it an extra classname to update its position if a service details is displayed 
- the detailed service stays like it is
I think it's difficult to handle right now because searchBar and services are not in the same container.
you'll tell me if it improves handling ;)
And no more pb with rows number !